R/summarize-data.R

```diff
@@ -134,7 +134,7 @@ summarize_data <- function(
     suppressMessages({
       sum <-
         data %>%
-        group_by(across(c(replicate, all_of(groups)))) %>%
+        group_by(across(c(all_of(replicate), all_of(groups)))) %>%
```

Comment:
At a glance, this looks correct to me (and it looks like there are tests that would light up if it weren't).

How you chose to write it is fine with me, but just a note that you could simplify this to use one `all_of` call:

``` r
library(tidyverse)

x <- c("gear", "cyl")
y <- "carb"

group_by(mtcars, across(c(all_of(x), all_of(y)))) %>%
  summarise(n = n(), .groups = "drop")
#> # A tibble: 12 × 4
#>     gear   cyl  carb     n
#>    <dbl> <dbl> <dbl> <int>
#>  1     3     4     1     1
#>  2     3     6     1     2
#>  3     3     8     2     4
#>  4     3     8     3     3
#>  5     3     8     4     5
#>  6     4     4     1     4
#>  7     4     4     2     4
#>  8     4     6     4     4
#>  9     5     4     2     2
#> 10     5     6     6     1
#> 11     5     8     4     1
#> 12     5     8     8     1

group_by(mtcars, across(all_of(c(x, y)))) %>%
  summarise(n = n(), .groups = "drop")
#> # A tibble: 12 × 4
#>     gear   cyl  carb     n
#>    <dbl> <dbl> <dbl> <int>
#>  1     3     4     1     1
#>  2     3     6     1     2
#>  3     3     8     2     4
#>  4     3     8     3     3
#>  5     3     8     4     5
#>  6     4     4     1     4
#>  7     4     4     2     4
#>  8     4     6     4     4
#>  9     5     4     2     2
#> 10     5     6     6     1
#> 11     5     8     4     1
#> 12     5     8     8     1
```